_playwright-tests/UI/SnapshotRepo.spec.ts

```diff
@@ -103,3 +103,117 @@ test.describe('Snapshot Repositories', () => {
     });
   });
 });
+
+test('Snapshot deletion', async ({ page }) => {
+  await navigateToRepositories(page);
+  await closePopupsIfExist(page);
+  const repoNamePrefix = 'snapshot-deletion';
+  const randomName = () => `${(Math.random() + 1).toString(36).substring(2, 6)}`;
+  const repoName = `${repoNamePrefix}-${randomName()}`;
+  const templateName = `Test-template-for-snapshot-deletion-${randomName()}`;
+
+  await test.step('Create a repository', async () => {
+    await page.getByRole('button', { name: 'Add repositories' }).first().click();
+    await expect(page.getByRole('dialog', { name: 'Add custom repositories' })).toBeVisible();
+    await page.getByLabel('Name').fill(`${repoName}`);
+    await page.getByLabel('Snapshotting').click();
+    await page.getByLabel('URL').fill('https://fedorapeople.org/groups/katello/fakerepos/zoo/');
+    await page.getByRole('button', { name: 'Save', exact: true }).click();
+    const row = await getRowByNameOrUrl(page, repoName);
+    await expect(row.getByText('Valid')).toBeVisible({ timeout: 60000 });
+  });
+
+  await test.step('Edit the repository', async () => {
+    for (let i = 2; i <= 4; i++) {
+      const row = await getRowByNameOrUrl(page, repoName);
+      await expect(row.getByText('Valid')).toBeVisible({ timeout: 60000 });
+      await test.step(`Edit repository and create snapshot ${i}`, async () => {
+        // Open the edit modal
+        await row.getByLabel('Kebab toggle').click();
+        await row.getByRole('menuitem', { name: 'Edit' }).click({ timeout: 60000 });
+        await page
+          .getByLabel('URL')
+          .fill(`https://fedorapeople.org/groups/katello/fakerepos/zoo${i}/`);
+        await page.getByRole('button', { name: 'Save changes', exact: true }).click();
+        await expect(row.getByText('Valid')).toBeVisible({ timeout: 60000 });
+      });
+    }
+  });
+
+  await test.step('Verify the snapshot count for the repo.', async () => {
+    const row = await getRowByNameOrUrl(page, repoName);
+    await expect(row.getByText('Valid')).toBeVisible({ timeout: 60000 });
+    await page.getByRole('button', { name: 'Kebab toggle' }).click();
+    await page.getByRole('menuitem', { name: 'View all snapshots' }).click();
+    // Count the number of rows in the snapshot list table
+    const snapshotCount = (await page.getByTestId('snapshot_list_table').locator('tr').count()) - 1; // Subtract 1 for the header row
+    // Create a template which uses the repo and assert that is uses the latest snapshot
+    await page.getByLabel('Close', { exact: true }).click();
+    await navigateToTemplates(page);
```

Comment:
the template creation could be a separate step 💅🏼 (i.e.: cut the verify count step after line 151 and make the template creation a separate step)